### PR TITLE
Extends dynamic-test-driver functionality | Adds new UPS Sender Endpoint

### PR DESCRIPTION
**Pull request by josemigallas**
### Dynamic-Test-Driver
Implements the functionality to read a CSV file with device tokens and send 1 push notification to each. Also to specify the delay between requests and to instantiate many test runners simultaneously.

All similar to the already implemented [rhmap-test-driver](https://github.com/aerogear/ups-test-suite/tree/master/rhmap-test-driver).

In order to verify it:
1. Start UPS up.
2. Create mocks with [mock-data-loader](https://github.com/aerogear/ups-test-suite/tree/master/mock-data-loader), saving them into a devices.csv file.
3. Run `node app/index.js -a <app id> -m <app master secret> -c ./devices.csv -d 10000 -i 5`
4. Verify that every 10 seconds there are 5 new notifications.

### RHMAP-Test-Driver
The Cloud App now has [a new route](https://github.com/josemigallas/rhmap-test-cloud-app/commit/4b046ae6ad5841fc76d221dca976fc60e3de2071) to invoke the UPS Sender API directly, without using `fh.push()`. The rhmap-test-driver has been upgraded to use this new route by adding the flag `-D`. It is also needed to specify the `pushApplicationID` and the `masterSecret` of the mobile application, that can be found under *push -> Sender API* in Studio.

### Logs
Logs have been improved in general. An ERROR should be an exception triggered by either the CloudApp or the TestDriver. Any other result will be considered a RESPONSE, regardless of it was successful or not.

### :warning: Yet More Changes
Added implementation to cover https://issues.jboss.org/browse/RHMAP-13659. Now by adding `-t` flag, all notifications are set via `sender.sendBatch()` method.

Also the command line parser has been extracted into a utility method.

### CODE REVIEW

**Comment by matzew** [comment]
Regarding the dynamic test driver, can we have the BASE_URL of the push server configured? 
e.g. to run against an instance, on openshift, or even RHMAP

**Comment by matzew** [comment]
Running it, with changed URL, I get this:
```
Total devices: 5000
Sending message to "undefined" {a333dafc-f74d-4d1b-83dc-a2fb973338f8}
Sending message to "undefined" {a333dafc-f74d-4d1b-83dc-a2fb973338f8}
Sending message to "undefined" {a333dafc-f74d-4d1b-83dc-a2fb973338f8}
RESPONSE: {}
RESPONSE: {}
Sending message to "undefined" {a333dafc-f74d-4d1b-83dc-a2fb973338f8}
RESPONSE: {}
Sending message to "undefined" {a333dafc-f74d-4d1b-83dc-a2fb973338f8}
RESPONSE: {}
Sending message to "undefined" {a333dafc-f74d-4d1b-83dc-a2fb973338f8}
RESPONSE: {}
Sending message to "undefined" {a333dafc-f74d-4d1b-83dc-a2fb973338f8}
RESPONSE: {}
Sending message to "undefined" {a333dafc-f74d-4d1b-83dc-a2fb973338f8}
RESPONSE: {}
Sending message to "undefined" {a333dafc-f74d-4d1b-83dc-a2fb973338f8}
```
but some messages seems to arrive at my UPS

**Comment by josemigallas** [comment]
@matzew The url is now configurable and regarding your last comment, I think that is the unique response `unifiedpush-node-sender` sends, regardless of any UPS error. With a wrong FCM URL and the real one is the same response :-1:

**Comment by matzew** [comment]
You mean the

```
RESPONSE: {}
```

is the 
> I think that is the unique response unifiedpush-node-sender sends ?

I don't think so, I think it reacts on error, of if UPS did return `202`:
https://github.com/aerogear/aerogear-unifiedpush-nodejs-client/blob/24fa6255b51c474c47cbc41059ccc263a62d7cdf/lib/sender.js#L123-L127

File: dynamic-test-driver/app/index.js [hunk 1]
```diff
@@ -1,16 +1,101 @@
...
-const username = process.argv[2];
-const password = process.argv[3];
+const DEFAULT_DELAY = 1000;
+const DEFAULT_INSTANCES = 1;
+const DEFAULT_ENDPOINT = "http://localhost:8080/ag-push";
 
-if (!username || !password) {
-    console.log("usage: node index.js <username> <password>");
-    process.exit(1);
+const args = argv
```
Comment (ziccardi) [inline dynamic-test-driver/app/index.js]: I think it would be better to put this into its own function (ie `const args = buildCommandLineParser();`)

**Comment by aliok** [comment]
I went through the code quickly. Looks ok to me.

I did the verification steps defined in PR description. I was able to see the messages on admin UI.

All good IMO.

@josemigallas great work!

**Review by matzew — APPROVED** [review]
:shipit:

**Pull request merged**